commit 5714aa740f30975b5fc51b1858fbca475a94c1d1
Author: Chuck Lever <chuck.lever@oracle.com>
Date:   Wed Feb 8 16:59:46 2017 -0500

    xprtrdma: Fix Read chunk padding
    
    commit 24abdf1be15c478e2821d6fc903a4a4440beff02 upstream.
    
    When pad optimization is disabled, rpcrdma_convert_iovs still
    does not add explicit XDR round-up padding to a Read chunk.
    
    Commit 677eb17e94ed ("xprtrdma: Fix XDR tail buffer marshalling")
    incorrectly short-circuited the test for whether round-up padding
    is needed that appears later in rpcrdma_convert_iovs.
    
    However, if this is indeed a regular Read chunk (and not a
    Position-Zero Read chunk), the tail iovec _always_ contains the
    chunk's padding, and never anything else.
    
    So, it's easy to just skip the tail when padding optimization is
    enabled, and add the tail in a subsequent Read chunk segment, if
    disabled.
    
    Fixes: 677eb17e94ed ("xprtrdma: Fix XDR tail buffer marshalling")
    Signed-off-by: Chuck Lever <chuck.lever@oracle.com>
    Signed-off-by: Anna Schumaker <Anna.Schumaker@Netapp.com>
    Signed-off-by: Greg Kroah-Hartman <gregkh@linuxfoundation.org>

Index: linuxtest/dummy/rpi_1349_5714aa740f30975b5fc51b1858fbca475a94c1d1.txt
===================================================================
--- /dev/null
+++ linuxtest/dummy/rpi_1349_5714aa740f30975b5fc51b1858fbca475a94c1d1.txt
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
